castervoice/lib/ccr/core/text_manipulation.py

```diff
@@ -0,0 +1,169 @@
+
+
+    
+
+        
+    mapping = {
+
+    # Todo: Find way to to better consolidate these context actions. 
+    # Todo: Put context actions for different apps based on pause time requirements
+    # Todo: Consolidate command definitions for left character versus right character; handle defaults in the functions, rather than choice objects.
+                
+        # PROBLEM: sometimes Dragon thinks the variables are part of dictation.           
+        
+        "replace <direction> [<number_of_lines_to_search>] [<occurrence_number>] <dictation> with <dictation2>":
+            R(Function(text_manipulation_functions.copypaste_replace_phrase_with_phrase,
+                       dict(dictation="replaced_phrase", dictation2="replacement_phrase")), 
+              rdescript="Text Manipulation: replace text to the left or right of the cursor"),
+        "replace <direction>  [<number_of_lines_to_search>] [<occurrence_number>] <character> with <character2>":
+            R(Function(text_manipulation_functions.copypaste_replace_phrase_with_phrase,
+                       dict(character="replaced_phrase", character2="replacement_phrase")), 
+              rdescript="Text Manipulation: replace character to the left of the cursor"),
+        
+        "remove <direction> [<number_of_lines_to_search>] [<occurrence_number>] <dictation>":
+            R(Function(text_manipulation_functions.copypaste_remove_phrase_from_text,
+                       dict(dictation="phrase")),
+                        rdescript="Text Manipulation: remove chosen phrase to the left or right of the cursor"),
+        "remove <direction> [<number_of_lines_to_search>] [<occurrence_number>] <character>":
+            R(Function(text_manipulation_functions.copypaste_remove_phrase_from_text,
+                       dict(character="phrase")), 
+              rdescript="Text Manipulation: remove chosen character to the left of the cursor"),
+        
+        "go <direction> [<number_of_lines_to_search>] [<before_after>] [<occurrence_number>] <dictation>":
+            R(Function(text_manipulation_functions.move_until_phrase,
+                       dict(dictation="phrase")), 
+               rdescript="Text Manipulation: move to chosen phrase to the left or right of the cursor"),
+        "go <direction> [<before_after>] [<number_of_lines_to_search>] [<occurrence_number>] <character>":
+            R(Function(text_manipulation_functions.move_until_phrase,
+                       dict(character="phrase")),
+              rdescript="Text Manipulation: move to chosen character to the left of the cursor"),
+
+        "grab <direction> [<number_of_lines_to_search>] [<occurrence_number>] <dictation>":
+            R(Function(text_manipulation_functions.select_phrase, 
+            dict(dictation="phrase")), 
+                 rdescript="Text Manipulation: select chosen phrase"),
+        "grab <direction> [<number_of_lines_to_search>] [<occurrence_number>] <character>":
```

Comment:
Remind me of the use case for "grab"? I haven't been using it nor come across a situation where one of the other functions wasn't more suitable.

Reply:
"Grab" is for select. Dragon always prioritizes its native commands "select", which is why I called it "grab" instead of "select". I have added some better comments in addition to the rdescript

Reply:
Nm cut and paste is a good enough use-case that I wasn't thinking about.